packages/flutter_tools/lib/src/dart/analysis.dart

```diff
@@ -46,7 +46,9 @@ class AnalysisDriver {
 
   DriverOptions options;
 
-  String get sdkDir => options.dartSdkPath ?? cli_util.getSdkDir().path;
+  String get sdkDir {
+    return options.dartSdkPath ?? path.absolute(cli_util.getSdkDir().path);
```

Comment:
There are other commands that use the dart SDK (e.g. format calls the dartfmt binary directly), but this seems to be the only one that uses internals of the Dart SDK, hence the slightly different call.